### PR TITLE
Fix argument exception messages and parameter names

### PR DESCRIPTION
**Pull request by iron9light**
A couple of exceptions were thrown with missing parameter names or using the exception message as the object name. This fixes both.

### CODE REVIEW

File: src/Microsoft.VisualStudio.Validation/Verify.cs [hunk 1]
```diff
@@ -126,9 +126,11 @@ public static void NotDisposed(IDisposableObservable disposedValue, string messa
         [DebuggerStepThrough]
         public static void NotDisposed([DoesNotReturnIf(false)] bool condition, object disposedValue, string message = null)
         {
+            Requires.NotNull(disposedValue, nameof(disposedValue));
```
Comment (iron9light) [inline src/Microsoft.VisualStudio.Validation/Verify.cs]: `public static void NotDisposed(IDisposableObservable disposedValue, string message = null)` requires `disposedValue` not null.
And we already has `public static void NotDisposed([DoesNotReturnIf(false)]bool condition, string message)`
So I think we should requires `disposedValue` not null here.

File: src/Microsoft.VisualStudio.Validation/Requires.cs [hunk 2]
```diff
@@ -169,7 +169,7 @@ public static void NotNullOrWhiteSpace([ValidatedNotNull]string value, string pa
 
             if (string.IsNullOrWhiteSpace(value))
             {
-                throw new ArgumentException(Format(Strings.Argument_Whitespace, parameterName));
+                throw new ArgumentException(Format(Strings.Argument_Whitespace, parameterName), parameterName);
```
Comment (drewnoakes) [inline src/Microsoft.VisualStudio.Validation/Requires.cs]: Consider adding a test that validates this change.

File: src/Microsoft.VisualStudio.Validation.Tests/VerifyTests.cs [hunk 3]
```diff
@@ -64,6 +63,17 @@ public void NotDisposed()
         {
             Assert.Equal(typeof(object).FullName, ex.ObjectName);
         }
+
+        try
+        {
+            Verify.NotDisposed(false, "message");
+            Assert.False(true, "Expected exception not thrown.");
+        }
+        catch (ObjectDisposedException ex)
```
Comment (drewnoakes) [inline src/Microsoft.VisualStudio.Validation.Tests/VerifyTests.cs]: You can simplify this and the blocks above, to get rid of the try/catch. `Assert.Throws` will return the caught exception, which you can then do your `Assert.Equals` on.

For example:

```c#
ObjectDisposedException ex = Assert.Throws<ObjectDisposedException>(() => Verify.NotDisposed(false, (object)null));
Assert.Equal(string.Empty, ex.ObjectName);
Assert.Equal("message", ex.Message);
```

File: src/Microsoft.VisualStudio.Validation/Verify.cs (hunk 1 above)
Comment (drewnoakes) [reply src/Microsoft.VisualStudio.Validation/Verify.cs]: Seems sensible, though a breaking change.

If accepted, there should be a test case that validates the ANE.

Comment (AArnott) [reply src/Microsoft.VisualStudio.Validation/Verify.cs]: That's a breaking change for a library that's used in a *lot* of places. I don't recall the reasoning for the difference in accepted `null` here, if there even was one. But this isn't one of those breaking changes that seem valuable to make in the face of possible costs. So I'm reverting this part of your change prior to merging.

File: src/Microsoft.VisualStudio.Validation.Tests/VerifyTests.cs (hunk 3 above)
Comment (AArnott) [reply src/Microsoft.VisualStudio.Validation.Tests/VerifyTests.cs]: Yes, I noticed the same thing. There's quite a bit of pattern precedent in this file so I appreciate that @iron9light. I'll follow up with a subsequent change to use the shorter pattern.

**Pull request merged**